### PR TITLE
Story/1543

### PR DESCRIPTION
**Pull request by joe-gb**
Adding functionality to unassign and cancel user batches on logout.

### CODE REVIEW

File: addons/udes_stock/static/src/js/logout.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    var ajax = require('web.ajax');
+    var Dialog = require('web.Dialog');
+
+    error = function(msg){
+        Dialog.alert(self, _(msg),{title: _("Error")})
+    }
+    logout_override = function(){
+        test = ajax.jsonRpc("/api/stock-picking-batch/check-user-batches/", 'call').then(function (data) {
+            if (data.error) {
+                error(data.error)
```
Comment (nocturnalastro) [inline addons/udes_stock/static/src/js/logout.js]: missing semicolon?

File: addons/udes_stock/static/src/js/logout.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+odoo.define('udes.stock.logout', function (require) {
+
+    var core = require('web.core');
+    var logout = core.action_registry.get("logout")
+    var ajax = require('web.ajax');
+    var Dialog = require('web.Dialog');
+
+    error = function(msg){
+        Dialog.alert(self, _(msg),{title: _("Error")})
```
Comment (nocturnalastro) [inline addons/udes_stock/static/src/js/logout.js]: missing semicolon?

File: addons/udes_stock/static/src/js/logout.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+odoo.define('udes.stock.logout', function (require) {
+
+    var core = require('web.core');
+    var logout = core.action_registry.get("logout")
```
Comment (nocturnalastro) [inline addons/udes_stock/static/src/js/logout.js]: missing semicolon

File: addons/udes_stock/static/src/js/logout.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+                    // Ask user to confirm logout
+                    Dialog.confirm(self, _("You still have work assigned - are you sure you wish to logout?"),{
+                        confirm_callback: function(){logout();},
+                        cancel_callback: function(){},
+                        title: _('Confirm logout'),
+                    });
+                }
+                else {
+                    // No batches, logout.
+                    logout()
```
Comment (nocturnalastro) [inline addons/udes_stock/static/src/js/logout.js]: semicolon

File: addons/udes_stock/static/src/js/logout.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+                }
+            }
+        });
+    }
+    core.action_registry.add('logout', logout_override);
+
+});
```
Comment (nocturnalastro) [inline addons/udes_stock/static/src/js/logout.js]: newline?

File: addons/udes_stock/static/src/js/logout.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+                else {
+                    // No batches, logout.
+                    logout();
+                }
+            }
+        });
+    }
+    core.action_registry.add('logout', logout_override);
+
+});
```
Comment (mcb30) [inline addons/udes_stock/static/src/js/logout.js]: I don't know enough about our front-end code to sensibly comment on this, sorry.

**Pull request merged**